src/Components/Patient/PatientRegister.tsx

```diff
@@ -1546,6 +1562,18 @@ export const PatientRegister = (props: PatientRegisterProps) => {
                                   />
                                 )}
                               </div>
+                              <SelectFormField
+                                {...field("occupation")}
+                                label="Occupation"
+                                placeholder="Select Occupation"
+                                options={occupationTypes}
+                                optionLabel={(
+                                  o: (typeof OCCUPATION_TYPES)[number]
+                                ) => o.text}
+                                optionValue={(
+                                  o: (typeof OCCUPATION_TYPES)[number]
+                                ) => o.id}
```

Comment:
types would be inferred automatically from the `options` prop.

```suggestion
                                optionLabel={(o) => o.text}
                                optionValue={(o) => o.id}
```